server/hotrod/src/main/java/org/infinispan/server/hotrod/configuration/HotRodServerConfigurationBuilder.java

```diff
@@ -125,6 +125,9 @@ public void validate() {
       if (proxyHost == null && host == null) {
          throw log.missingHostAddress();
       }
+      if (proxyPort < 0 && port == 0 && !startTransport) {
```

Comment:
@pruivo The (reverse) proxy has to know about the server's port in order to forward requests, I'm not sure how that would work if the server port changes on each restart.